ros_gz_sim/src/ros_gz_bridge_system.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <memory>
+#include <string>
+#include <thread>
+
+#include <gz/common/Util.hh>
+#include <gz/plugin/Register.hh>
+#include <rclcpp/rclcpp.hpp>
+#include <ros_gz_bridge/bridge_config.hpp>
+#include <ros_gz_bridge/ros_gz_bridge.hpp>
+
+#include "ros_gz_bridge_system.hpp"
+
+namespace ros_gz_sim
+{
+/// \brief Private ROSGzBridgeSystem data class.
+class ROSGzBridgeSystemPrivate
+{
+public:
+  /// \brief The ROS 2 <--> Gz bridge.
+  std::shared_ptr<ros_gz_bridge::RosGzBridge> bridge;
+
+  /// \brief The ROS 2 executor.
+  std::shared_ptr<rclcpp::executors::MultiThreadedExecutor> exec;
+
+  /// \brief A thread to call spin and not block the Gazebo thread.
+  std::thread thread;
+};
+
+//////////////////////////////////////////////////
+ROSGzBridgeSystem::ROSGzBridgeSystem()
+: System(), dataPtr(new ROSGzBridgeSystemPrivate())
+{
+}
+
+//////////////////////////////////////////////////
+ROSGzBridgeSystem::~ROSGzBridgeSystem()
+{
+  if (this->dataPtr->exec) {
+    this->dataPtr->exec->cancel();
+    this->dataPtr->thread.join();
+  }
+}
+
+//////////////////////////////////////////////////
+void ROSGzBridgeSystem::Configure(
+  const gz::sim::Entity & /*_entity*/,
+  const std::shared_ptr<const sdf::Element> & _sdf,
+  gz::sim::EntityComponentManager & /*_ecm*/,
+  gz::sim::EventManager & /*_eventMgr*/)
+{
+  // Ensure that ROS is setup.
+  if (!rclcpp::ok()) {
+    rclcpp::init(0, nullptr);
+  }
+
+  if (!_sdf->HasElement("config_file")) {
+    std::cerr << "No <config_file> found. Plugin disabled." << std::endl;
```

Comment:
as this is a Gazebo plugin, should we use here `gzerr` ? remove `<iostream>` header too

Reply:
Changed in [78067c4](https://github.com/gazebosim/ros_gz/pull/511/commits/78067c419f272ce86ec1a3060929f4fd037afabe).